packages/x-adapter/README.md

```diff
@@ -1,13 +1,456 @@
+`RequestOptions.endpoint` parameter in the function call. Take in account that your `responseMapper`
+definition should be agnostic enough to support the change:
+
+```ts
+export const getItemById = endpointAdapterFactory({
+  endpoint: 'https://dummyjson.com/quotes/{id}',
+  // ... rest of options to configure
+  }
+});
+
+// You would pass the new endpoint in the function call
+getItemById({ id: 1 }, { endpoint: 'https://dummyjson.com/products/{id}');
+```
+
+<br>
+
+#### Using the httpClient function
+
+The `x-adapter` uses the `Fetch API` interface by default to perform the requests. But you can use
+your own HTTP Client as part of the configurable `EndpointAdapterOptions`. The `httpClient` is a
+function type that accepts two parameters: the `endpoint` and an additional
+[`options`](https://github.com/empathyco/x/blob/main/packages/x-adapter/src/http-clients/types.ts)
+object to make the request with.
+
+```ts
+// HTTP Client
+const customHttpClient: HttpClient = (endpoint, options) =>
+  axios.get(endpoint, { params: options?.parameters }).then(response => response.data);
+
+// Request Mapper
+const customRequestMapper: Mapper<AppSearchRequest, ApiRequest> = ({ query }) => {
+  return {
+    q: query
+  };
+};
+
+// Object Mapper to use in the Response Mapper
+const productMap: Mapper<ApiProduct, AppProduct> = product => {
+  return {
+    id: product.id.toString(),
+    name: product.title,
+    price: product.price
+  };
+};
+
+// Response Mapper
+const customResponseMapper: Mapper<ApiSearchResponse, AppSearchResponse> = ({
+  products,
+  total
+}) => {
+  return {
+    products: products.map(product => productMap(product, {})),
+    total: total
+  };
+};
+
+// Adapter factory function implementation
+export const searchProducts = endpointAdapterFactory({
+  endpoint: 'https://dummyjson.com/products/search',
+  httpClient: customHttpClient,
+  requestMapper: customRequestMapper,
+  responseMapper: customResponseMapper
+});
+```
+
+<br>
+
+### Implement your own adapter using schemas
+
+The `x-adapter` library includes a `schemaMapperFactory` function that returns a mapper function to
+execute the `schema` you provide. The function accepts a `Source` object (the data to be
+transformed) and a `Target` object. You just need to create the request and response models to
+define the structure of your data vs the APIs data, and pass them to the factory function.
+
+###### Types definition
+
+```ts
+// API data models
+interface ApiUserRequest {
+  q: string;
+}
+interface ApiUserResponse {
+  users: ApiUser[];
+  total: number;
+}
+interface ApiUser {
+  id: number;
+  firstName: string;
+}
+
+// App's data models
+interface AppUserRequest {
+  query: string;
+}
+interface AppUserResponse {
+  people: AppUser[];
+  total: number;
+}
+interface AppUser {
+  id: string;
+  name: string;
+}
+```
+
+###### Schema's mapper factory function implementation
+
+```ts
+// Object Mapper to use in the Response Mapper
```

Comment:
Like above, I would just inline this mapper inside the `userSearchResponseMapper` `users.map(...)` call.